selfservice/strategy/oidc/provider_generic_oidc.go

```diff
@@ -83,8 +83,13 @@ func (g *ProviderGenericOIDC) AuthCodeURLOptions(r ider) []oauth2.AuthCodeOption
 	return options
 }
 
-func (g *ProviderGenericOIDC) verifyAndDecodeClaimsWithProvider(ctx context.Context, provider *gooidc.Provider, raw string) (*Claims, error) {
-	token, err := provider.Verifier(&gooidc.Config{ClientID: g.config.ClientID}).Verify(ctx, raw)
+func (g *ProviderGenericOIDC) verifyAndDecodeClaimsWithProvider(ctx context.Context, provider *gooidc.Provider, rawIdToken string) (*Claims, error) {
+	token, err := provider.
+		Verifier(&gooidc.Config{
+			ClientID:          g.config.ClientID,
+			SkipClientIDCheck: g.config.SkipAudienceCheck,
```

Comment:
This is dangerous and we need to remove this feature for the same reason we need to remove the `ClaimsFromAccessToken` one.

```suggestion
```

Reply:
done